framework/contracts/native/ans-host/src/contract.rs

```diff
@@ -103,13 +103,29 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
 }
 
 #[cfg_attr(feature = "export", cosmwasm_std::entry_point)]
-pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> AnsHostResult {
-    let version: Version = CONTRACT_VERSION.parse().unwrap();
+pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> AnsHostResult {
+    match msg {
+        MigrateMsg::Instantiate(instantiate_msg) => {
```

Comment:
Love that logic hah. Instantiate on Migrate is fun

Reply:
We should make sure that instantiate is callable only once I think !

Reply:
> We should make sure that instantiate is callable only once I think !

Yeah we can add check that there is no cw2 in state. Is it good enough?